lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -113,7 +114,21 @@ def add_files_to_build_phases
       def add_resources_bundle_targets
         target.file_accessors.each do |file_accessor|
           file_accessor.resource_bundles.each do |bundle_name, paths|
-            file_references = paths.map { |sf| project.reference_for_path(sf) }
+            file_references = paths.map do |path|
+              ref = project.reference_for_path(path)
+
+              # Some nested files are not directly present in the Xcode project, such as the contents
+              # of an .xcdatamodeld directory. These files are implicitly included by including their
+              # parent directory.
+              next nil if ref.nil?
+
+              # For variant groups, the variant group itself is added, not its members.
+              next ref.parent if ref.parent.is_a?(Xcodeproj::Project::Object::PBXVariantGroup)
```

Comment:
No need. Variant groups can only have direct children. Possible edge cases and how they're handled:

- Core Data models: These can't be localized. It's possible there's a way to get this to happen by manipulating the project XML just right, but the Xcode UI doesn't support it and I bet it would break in other ways. Also, conceptually it doesn't make any sense.
- Asset Catalogs: Conceptually this might make sense, but Xcode doesn't support it in the UI. Also, I believe this would actually just work since Asset Catalogs are just a PBXFileReference for the top of the directory in the project structure.
- Localized directories (`en.lproj/images/image.png` and `fr.lproj/images/image.png`): The "images" directory is what actually gets localized here. Xcode never has any notion of "image.png". This should work fine for things like `UIImage.image(named:"images/image.png")` (although the `imageNamed:` documentation explicitly says don't do this). For strings and interface files they would fail to be compiled since Xcode doesn't know about them directly. There's no supported way to do this in Xcode today. We could possibly construct file references to the images and nest them multiple levels up:
```
  Variant Group "image.png"
     en.lproj/images/image.png
     fr.lproj/images/image.png
```
This would potentially work, but: (a) there's no way to do this in Xcode today, so it's likely to break, (b) this is a total edge case that can be solved in better ways (just remove the middle directory), and (c) this would still actually work for L126 above since the variant group would still be the direct parent of the image file references in the project structure.


There short version is that there are no cases that Xcode supports today where variant groups are allowed to contain other groups, so it's safe to just check one level up.